internal/rbd/rbd_util.go

```diff
@@ -2038,12 +2041,28 @@ func (ri *rbdImage) listSnapAndChildren() ([]librbd.SnapInfo, []string, error) {
 		return nil, nil, err
 	}
 
-	// ListChildren() returns pools, images, err.
-	_, children, err := image.ListChildren()
+	// ListChildrenAttributes() returns child ImageSpec list, err.
+	childImageSpecList, err := image.ListChildrenAttributes()
 	if err != nil {
 		return nil, nil, err
 	}
 
+	children := make([]string, 0)
+	volSnapChildren := make([]string, 0)
+	for _, child := range childImageSpecList {
+		if child.Trash {
+			continue
+		}
+		if strings.HasSuffix(child.ImageName, "-temp") {
+			children = append(children, child.ImageName)
+		} else {
+			volSnapChildren = append(volSnapChildren, child.ImageName)
+		}
+	}
+	// order the temp clone images first followed by the volume snapshots
+	// so that the temp clones are flattened first.
```

Comment:
This is the 2nd comment about `temp clones are flattened first`. I don't get why that is the case. If that is a requirement of the returned list, mention something about the order. Or wherever the list is consumed, use some other way of inspecting the list. Might be cleaner to return two lists, so that the distinction is very clear.

Reply:
> This is the 2nd comment about `temp clones are flattened first`. I don't get why that is the case. If that is a requirement of the returned list, mention something about the order. Or wherever the list is consumed, use some other way of inspecting the list. Might be cleaner to return two lists, so that the distinction is very clear.

I've removed it from the function definition, since we still are returning full list of child images which are not in trash.

I've expanded the comment in this section with reasons why such ordering is done so it can be understood in the future.

Reply:
Thanks, that makes things a little bit clearer. I don't understand (yet) if, and why callers need to care about the order in the list that is returned. If callers need to care, or benefit from it, why not return separate lists? Or, a struct like `{ snaps, tempChildren, volSnapChildren }` so that callers of the function are very aware of the differences, and can decide how to handle them?

Reply:
> Thanks, that makes things a little bit clearer. I don't understand (yet) if, and why callers need to care about the order in the list that is returned. If callers need to care, or benefit from it, why not return separate lists? Or, a struct like `{ snaps, tempChildren, volSnapChildren }` so that callers of the function are very aware of the differences, and can decide how to handle them?

I've moved the logic to order images into the caller and added a struct for returned parameters.
PTAL